### PR TITLE
`av stack swtich` supports a branch or PR URL as arg

### PR DESCRIPTION
**Pull request by Konboi**
Address: https://github.com/aviator-co/av/issues/325

### CODE REVIEW

File: cmd/av/stack_switch.go [hunk 1]
```diff
@@ -99,6 +117,52 @@ func stackSwitchBranchList(repo *git.Repo, tx meta.ReadTx, branches map[string]*
...
+	if branch, err := parsePullRequestURL(input); err == nil {
+		return branch, nil
+	}
+
+	return input, nil
+}
+
+var PULL_REQUEST_URL_REGEXP = regexp.MustCompile(`^/([^/]+)/([^/]+)/pull/(\d+)`)
+
+func parsePullRequestURL(prURL string) (string, error) {
```
Comment (Konboi) [inline cmd/av/stack_switch.go:130]: I referred https://github.com/cli/cli/blob/f972050dc92c435c75b6ae7559592bfe28fa3dbd/pkg/cmd/pr/shared/finder.go#L209

File: cmd/av/stack_switch.go [hunk 2]
```diff
@@ -99,6 +117,52 @@ func stackSwitchBranchList(repo *git.Repo, tx meta.ReadTx, branches map[string]*
...
+
+	prID, err := strconv.Atoi(m[3])
+	if err != nil {
+		return "", errors.Wrap(err, "failed to parse pull request ID")
+	}
+	pr, err := client.GetPullRequest(context.Background(), gh.PullRequestOpts{
+		Owner:  m[1],
+		Repo:   m[2],
+		Number: int64(prID),
+	})
```
Comment (draftcode) [inline cmd/av/stack_switch.go:158]: You can find the branch based on https://github.com/aviator-co/av/blob/d74f1e1103e0e7e2eef3f8f3c7a3a08c38e81aa2/internal/meta/branch.go#L22. This shouldn't query GitHub.

Comment (Konboi) [reply cmd/av/stack_switch.go]: Does this mean that this [AllBranches method](https://github.com/aviator-co/av/pull/360/files#diff-1421442c9c429b4ddaaf38864ee227f4960ec5be73dcbcbd2031f69f92acfef2R148) also holds remote branch information locally?

Comment (Konboi) [reply cmd/av/stack_switch.go]: fixed https://github.com/aviator-co/av/pull/360/commits/6f8cc4a349f1850dd71cd0fea9ff385db37bbde8

Comment (draftcode) [reply cmd/av/stack_switch.go]: The local metadata DB should keep those PR info.

**Review by draftcode — APPROVED** [review]

**Pull request merged**